homeassistant/components/kraken/__init__.py

```diff
@@ -62,11 +60,10 @@ def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
         self._hass = hass
         self._config_entry = config_entry
         self._api = pykrakenapi.KrakenAPI(krakenex.API(), retry=0, crl_sleep=0)
-        self.tradable_asset_pairs = None
-        self.coordinator = None
-        self.unsub_listeners = []
+        self.tradable_asset_pairs: dict[str, str] = {}
+        self.coordinator: DataUpdateCoordinator | None = None
```

Comment:
It's return type of `async_update`, you can create an alias for it if you want.
```suggestion
        self.coordinator: DataUpdateCoordinator[dict[str, dict[str, float]] | None] | None = None
```